source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -233,38 +233,52 @@ void ShaderGraph::addDefaultGeomNode(ShaderInput* input, const GeomProp& geompro
     input->makeConnection(node->getOutput());
 }
 
-void ShaderGraph::addColorTransformNode(ShaderOutput* output, const string& colorTransform, ShaderGenerator& shadergen)
+void ShaderGraph::addColorTransformNode(ShaderInput* input, const ColorSpaceTransform& transform, ShaderGenerator& shadergen)
 {
-    const string nodeDefName = "ND_" + colorTransform + "_" + output->type->getName();
-    NodeDefPtr nodeDef = _document->getNodeDef(nodeDefName);
-    if (!nodeDef)
+    input->node->_classification |= ShaderNode::Classification::DO_NOT_OPTIMIZE;
```

Comment:
I like the idea of being able to tag nodes as "hands off" for the optimizer. In the case of constant nodes though we could actually remove them (since they are just a passthrough if a color transform is connected upstream) and instead plug the transform node directly to the downstream input. 
This is probably for another change list though, I just wanted to mention it.

Reply:
Looking closer I realise the optimizer is run before the color transform nodes are injected, so when/if removing constant nodes we need to pass any colorspace attribute along downstream to make this input use the correct colorspace. But again not really part of this change list.

Reply:
Although, for this specific use of DO_NOT_OPTIMIZE, it means that any node that has a color managed input cannot be optimized. To make this more future proof maybe only add the DO_NOT_OPTIMIZE flag if the node is a constant node?

Reply:
This ok?
```
if (input->node->hasClassification(ShaderNode::Classification::CONSTANT))
{
    input->node->_classification |= ShaderNode::Classification::DO_NOT_OPTIMIZE;
}
```

Reply:
Looks good! And maybe add a comment that we can improve on this later, replacing the constant node with the transform node.